portal/components/table/index.tsx

```diff
@@ -0,0 +1,178 @@
+'use client'
+
+import {
+  ColumnDef,
+  flexRender,
+  getCoreRowModel,
+  useReactTable,
+} from '@tanstack/react-table'
+import { useWindowSize } from 'hooks/useWindowSize'
+import { useRef } from 'react'
+import Skeleton from 'react-loading-skeleton'
+
+import { useInfiniteScroll } from './_hooks/useInfiniteScroll'
+import { useScrollbarDetection } from './_hooks/useScrollbarDetection'
+import { useTableData } from './_hooks/useTableData'
+import { Column, ColumnHeader } from './table'
+
+export type TableProps<T> = {
+  columns: ColumnDef<T, unknown>[]
+  data?: T[]
```

Comment:
[Nitpick] I think data should always be defined. It's a needed prop for a Table. However, it's true it can be `undefined` in some cases. Because of that, I think using 


```suggestion
  data: T[] | undefined
```

better represents the desired API. What do you think=